markdown/src/commonMain/kotlin/com/mohamedrajeb/richeditor/markdown/MarkdownImport.kt

```diff
@@ -0,0 +1,345 @@
+            )
+        }
+
+        MarkdownElementTypes.IMAGE -> {
+            val link = node.findChildOfType(MarkdownElementTypes.LINK_DESTINATION)?.getTextInNode(content)
+                ?.toString()
+            //todo add image
+        }
+
+        MarkdownElementTypes.LINK_DEFINITION -> {
+            val linkLabel =
+                node.findChildOfType(MarkdownElementTypes.LINK_LABEL)?.getTextInNode(content)?.toString()
+            if (linkLabel != null) {
+                val destination =
+                    node.findChildOfType(MarkdownElementTypes.LINK_DESTINATION)?.getTextInNode(content)
+                        ?.toString()
+                addToLastOrCreateNew(
+                    RichSpan(
+                        text = linkLabel,
+                        paragraph = getLastOrNewParagraph(),
+                        spanStyle = MarkdownSpanStyles.linkSpanStyle,
+                        style = RichSpanStyle.Link(destination ?: "")
+                    )
+                )
+            }
+        }
+
+        MarkdownTokenTypes.EOL -> appendText('\n')
+//        MarkdownTokenTypes.HARD_LINE_BREAK -> appendText("\n")
+        else -> handled = false
+    }
+    return handled
+}
+
+private fun ASTNode.listItems(
+    parser: Parser,
+    content: String,
+    level: Int,
+    listItem: ASTNode.(index: Int) -> Unit,
+) {
+    var index = -1
+    for (child in children) {
+        index++
+        when (child.type) {
+            MarkdownElementTypes.LIST_ITEM -> {
+                child.listItem(index)
+                if (child.children.isNotEmpty()) {
+                    child.listItems(
+                        parser,
+                        content = content,
+                        level = level,
+                        listItem,
+                    )
+                }
+            }
+
+            MarkdownElementTypes.ORDERED_LIST -> child.orderedList(
+                parser = parser,
+                content = content,
+                level = level + 1
+            )
+
+            MarkdownElementTypes.UNORDERED_LIST -> child.unorderedList(
+                parser = parser,
+                content = content,
+                level = level + 1
+            )
+        }
+    }
+}
+
+internal fun ASTNode.unorderedList(
+    parser: Parser,
+    content: String,
+    level: Int = 0
+) {
+    val paragraph = RichParagraph(
+        type = RichParagraph.Type.UnorderedList
+    )
+//    print("UNORDERED_LIST->")
+    listItems(
+        parser = parser,
+        content = content,
+        level = level
+    ) { listItemIndex ->
+
+        fun children(): MutableList<RichSpan> {
+//            print("LISTITEM->")
+            return parser.buildSpans(
+                content = content,
+                parent = paragraph,
+                children = children.filter { it.type != MarkdownTokenTypes.EOL }
+            )
+        }
+
+        fun appendTextListItem() {
+            paragraph.children.add(
```

Comment:
I am adding the list item child here , It should work but its not working , I added line breaks to separate the rich spans but it still didn't display right